### DISCUSSION TITLE
sftpgo portable

### DISCUSSION THREAD

**Discussion by mahirkemal** [drakkan/sftpgo#320] [Q&A] [answered]
Hi,

I am using the latest docker image but I could not configure sftp users from container CLI via sftpgo portable. 

I am running the below command:
**sftpgo portable --fs-provider 0 --sftp-username "abcdef" --sftp-password "12345" --sftpd-port "2022" permissions "*"**

Given user does not appear on web GUI and I am not able to do sftp operation via this user.

Am I missing something here? Is there any way to add new sftp users via CLI?

Thanks,
Mahir

**Comment by drakkan** [comment, marked as answer]
Hi,

the portable mode allows to share a single directory on demand, it creates a temporary user and the user will not be stored. You can use this user until the portable command is running.

If you want to load initial data please take a look at the `--loaddata-from` serve argument [here](https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md). You can create a dump from the web ui (maintenance section).

If instead you really want to create users from CLI you can use the [REST API](https://github.com/drakkan/sftpgo/blob/main/docs/rest-api.md). Some generators allow to create bash clients that use `curl` under the hood.

**Reply by mahirkemal** [reply]
Thank you for your quick response. 

My understanding is that we are able to use configured sftp_username & sftp_password until the next portable command run. Actually, thats what I wanted but I could not do sftp ops with the configured user credentials as I said. 

Also, I just want to pass my SFTP username and password parameters when running docker run. I've tried to do this by passing environment variables but there are no sftpd username and password variables within the default config file.

I will try the --loaddata-from and REST API options.

Thanks,
Mahir

**Reply by drakkan** [reply]
> Thank you for your quick response.
> 
> My understanding is that we are able to use configured sftp_username & sftp_password until the next portable command run. Actually, thats what I wanted but I could not do sftp ops with the configured user credentials as I said.
> 

This works for me:

```
docker run --name some-sftpgo -p 2022:2022 "drakkan/sftpgo:edge" sftpgo portable --username "abcdef" --password "12345" --sftpd-port "2022" --permissions "*"
```

`sftp_username` and `sftp_password` are for the sftp provider (use another sftp server as backend instead of the local storage)

> Also, I just want to pass my SFTP username and password parameters when running docker run. I've tried to do this by passing environment variables but there are no sftpd username and password variables within the default config file.
> 
> I will try the --loaddata-from and REST API options.
> 
> Thanks,
> Mahir

**Reply by mahirkemal** [reply]
Great! That's worked. Thanks much for your help.
